test/jdk/java/awt/print/PrinterJob/LandscapeStackOverflow.java

```diff
@@ -43,6 +45,10 @@
 
     public static final void main( String[] parameters ) {
         PrinterJob printjob = PrinterJob.getPrinterJob();
+        PrintService defaultPrtSrv = PrintServiceLookup.lookupDefaultPrintService();
+        if (printjob.getPrintService() == null || defaultPrtSrv == null) {
+            throw new RuntimeException("No printer found");
```

Comment:
We need to make this more robust.
The checks should be before we create the pritner job and should be to find ANY printer and none
throws the excepton, but if there is only a non-default one, set that as the printer for the job after creating the job.

That may not matter on Windows but on other platforms I am sure you can have printers, but not a default one.

Reply:
I was of the understanding that if there are multiple, then we have to choose one as default so defaultPrintService will not be null. But, the above code looks more apt.